trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/FunctionHandler.java

```diff
@@ -396,17 +400,20 @@ public Result handleFunctionDefinition(final IDispatcher main, final MemoryHandl
 		return new Result(impl);
 	}
 
-	private static CFunction updateVarArgsUsage(final IASTFunctionDefinition node, final CFunction oldFunType) {
-		final CFunction funType;
+	private static CFunction updateVarArgsUsage(final ILocation loc, final IASTFunctionDefinition node,
+			final CFunction funType, final String funName) {
 		// update varags usage
-		if (oldFunType.hasVarArgs() && oldFunType.getVarArgsUsage() == VarArgsUsage.UNKNOWN) {
+		if (funType.hasVarArgs() && funType.getVarArgsUsage() == VarArgsUsage.UNKNOWN) {
 			// if the function body creates a va_list object it uses its varargs
-			final ASTNameCollector vaListFinder = new ASTNameCollector("va_list");
+			final ASTNameCollector vaListFinder = new ASTNameCollector("__builtin_va_start");
 			node.getBody().accept(vaListFinder);
-			final boolean usesVarArgs = vaListFinder.getNames().length > 0;
-			funType = oldFunType.updateVarArgsUsage(usesVarArgs);
-		} else {
-			funType = oldFunType;
+			final int numberOfVaStarts = vaListFinder.getNames().length;
+			if (numberOfVaStarts > 1) {
+				// TODO: This requires a different handling of va_start and va_end.
+				// Currently they are simply handled by assignment and deallocation.
+				throw new UnsupportedSyntaxException(loc, funName + " has multiple calls to va_start.");
+			}
+			return funType.updateVarArgsUsage(numberOfVaStarts != 0);

```

Comment:
Aren't there also other cases where varargs are used, e.g. by only passing them on to another call?